### PR TITLE
make RSA and DSA operations throw MISSING_PRIVATE_KEY if needed, adapt ECDSA

### PR DESCRIPTION
**Pull request by DDvO**
This fixes issue #9457.

### CODE REVIEW

File: crypto/dsa/dsa_ossl.c [hunk 1]
```diff
@@ -126,6 +126,10 @@ static DSA_SIG *dsa_do_sign(const unsigned char *dgst, int dlen, DSA *dsa)
     BN_set_flags(tmp, BN_FLG_CONSTTIME);
 
     /* tmp := blind * priv_key * r mod q */
+    if (dsa->priv_key == NULL) {
+        reason = DSA_R_MISSING_PRIVATE_KEY;
+        goto err;
+    }
```
Comment (mattcaswell) [inline crypto/dsa/dsa_ossl.c]: I'd suggest moving this right up to near the check for other missing parameters: p, q and g

**Comment by mattcaswell** [comment]
Should this be backported to 1.1.1?

File: crypto/dsa/dsa_ossl.c (hunk 1 above)
Comment (DDvO) [reply crypto/dsa/dsa_ossl.c]: Done

**Comment by romen** [comment]
uhm, there was a commit after @mattcaswell approval: it seems a trivial change but I will wait until he reconfirms before merging

**Comment by romen** [comment]
This cherry-picks cleanly to 1.1.1

@mattcaswell although you suggested it, formally there was no 1.1.1 label on this when you approved, can you reconfirm once more?

Also as an editorial comment, for both master and 1.1.1, I plan to squash the two commits of this PR into a single one upon final merge.

**Comment by romen** [comment]
- [x] Merged in `master` with 7408f6759f1b0100438ca236ea8f549454aaf2d5
- [x] Merged in `1.1.1-stable` with ffc2b6373aabcdcfbb0ac725a00a907834202c4f

Thanks @DDvO !

**Pull request closed without merging**